rpg.py

```diff
@@ -41,56 +54,136 @@ def deal_damage(self, other_player):
+        """returns the location of all enemy players in a list"""
         return [
             index for index, player in enumerate(players)
             if not self.is_ally(player) and player.is_alive()
         ]
 
+    def get_all_allies(self, players: List) -> List[int]:
+        """returns the locations of all ally players in a list"""
+        return[
+            index for index, player in enumerate(players)
+            if self.is_ally(player1) and self.is_alive()
+        ]
+
     def act(self, players: List) -> List:
+        """how one player acts upon another, eg. an attack"""
         all_enemy_locations = self.get_all_enemies(players)
         if all_enemy_locations:
             if self.is_alive:
-                print(f'{self.name}\'s turn!')
-            targeted_index = all_enemy_locations[0]
-            targeted_player = players[targeted_index]
-            damaged_player = self.deal_damage(targeted_player)
-            players[targeted_index] = damaged_player
+                targeted_index = all_enemy_locations[0]
+                targeted_player = players[targeted_index]
+                damaged_player = self.deal_damage(targeted_player)
+                players[targeted_index] = damaged_player
         return players
 
 
 class Ally(Character):
     # subclass of character
     """This is a subclass of characters specific to team 1"""
-    def __init__(self, name: str, hp: int, attack: int, speed: int):
-        super().__init__(name, hp, attack, speed, team=1, level=1, exp=0, target_exp=200)
+    def __init__(self,
+                 name: str,
+                 hp: int,
+                 max_hp: int,
+                 attack: int,
+                 speed: int):
+        super().__init__(name,
+                         hp,
+                         max_hp,
+                         attack,
+                         speed,
+                         team=1,
+                         level=1,
+                         exp=0,
+                         target_exp=200)
 
     def act(self, players):
         """the act method specific to team 1"""
-        # I want to input the name of the enemy to attack
+
         all_enemy_locations = self.get_all_enemies(players)
         enemies = {character.name:
                    character for character in players if character.team != 1}
         if all_enemy_locations:
             if self.is_alive:
-                print(f'{self.name}\'s turn!')
-                print('Who do you attack?')
-                for player in players:
-                    if not player.is_ally(self) and player.is_alive():
-                        print(player.name)
-                targeted_enemy = None
-                while targeted_enemy is None:
-                    user_input = input('>')
-                    targeted_enemy = enemies.get(user_input)
-                damaged_player = self.deal_damage(targeted_enemy)
+                if all_enemy_locations.count(1):
```

Comment:
https://docs.python.org/3/tutorial/datastructures.html
You want `len()`
This counts how many times `1` appears in `all_enemy_locations`